framework_tvb/tvb/interfaces/web/static/js/filters.js

```diff
@@ -124,44 +123,15 @@ function refreshData(parentDivId, divSufix, name, sessionStoredTreeKey, gathered
         return;
     }
 
-    // This argument is required by the server.
-    // If absent set a falsy default as updateDivContent() checks for it. Has to be a string because it's in the url.
-    if (parentDivId === ""){
-        parentDivId = " ";
-    }
     //Make a request to get new data
     doAjaxCall({
-        async: false,  //todo: Is this sync really needed? It slows down the page.
-        type: 'GET', //todo: why is this a get? a post with the json seems better.
-        url: "/flow/getfiltereddatatypes/" + name + "/" + parentDivId + '/' + sessionStoredTreeKey + '/' + $.toJSON(gatheredData),
-        success: function (r) {
-            var elements = document.getElementsByName(name);
-            //Look for the previous select input whose data needs to be refreshed
-            if (elements.length > 1) {
-                //If more than one was found it's because of multiple algorithms
-                //We need to get only the one corresponding to the current algorithm
-                for (var i = 0; i < elements.length; i++) {
-                    var parent = elements[i].parentNode;
-                    //In case more components with the same name exist look for the
-                    //parent's div id
-                    while (parent.id == '' || parent.id == null) {
-                        parent = parent.parentNode;
-                    }
-                    if (divId != null && divId.indexOf(parent.id) !== -1) {
-                        //Remove the childs from this div and then recreate the components
-                        //using the html returned by the ajax call
-                        replaceSelect(elements[i].parentNode, r, name);
-                    }
-                }
-            } else if (elements.length === 1) {
-                replaceSelect(elements[0].parentNode, r, name);
-            } else {
-                displayMessage("Filter could not be applied!" + name, "infoMessage");
-                return;
-            }
-            displayMessage("Filters applied...");
+        type: 'POST',
+        url: "/flow/get_filtered_datatypes/" + name + '/' + datatypeIndex + '/' + $.toJSON(gatheredData),
+        success: function (response) {
+            var select_field = document.getElementById(name);
+            select_field.innerHTML = response;
```

Comment:
Basically here compose the options (from JSON response), if they can not be returned by a Jinja template

Reply:
I think I can't have a JSON response without using some template, when I tried to return for example {'options': string_of_options} only the string options appeared in the response.

Reply:
in this case (when the return value is a json) see to have the correct annotation(s) on the controller method also...
there are multiple examples in the code already and working

Reply:
if you choose annotation `@ using_template` in the controller then you expect a HTML string here (as you do right now) to put directly as innerHTML. If you choose  `@ jsonify` in the controller, then in the JS would be something like:
` r = $.parseJSON(r);
    if (r['status'] === 'ok') ...
`